analysis_driver/pipelines/qc_gatk4.py

```diff
@@ -0,0 +1,643 @@
+
+    input_vcf = Parameter()
+
+    def _run(self):
+        select_var_command = self.gatk_cmd('SelectVariants', self.raw_indels_vcf)
+        select_var_command += ' -V ' + self.input_vcf
+        select_var_command += ' -select-type INDEL '
+        select_variants_status = executor.execute(
+            select_var_command,
+            job_name='indel_select',
+            working_dir=self.exec_dir,
+            mem=16
+        ).join()
+        return select_variants_status
+
+
+class SNPsFiltration(GATK4Stage):
+    """Filter the SNPs using generic Hard filter."""
+
+    def _run(self):
+        filter_array = [
+            'QD < 2.0',
+            'FS > 60.0',
+            'MQ < 40.0',
+            'MQRankSum < -12.5',
+            'ReadPosRankSum < -8.0',
+            'SOR > 3.0'
+        ]
+        filters = "'" + ' || '.join(filter_array) + "'"
+        var_filter_command = self.gatk_cmd('VariantFiltration', self.hard_filtered_snps_vcf)
+        var_filter_command += " -V " + self.raw_snps_vcf
+        var_filter_command += " --filter-expression " + filters
+        var_filter_command += " --filter-name 'SNP_HARD_FILTER'"
+        variant_filter_status = executor.execute(
+            var_filter_command,
+            job_name='snps_filtration',
+            working_dir=self.exec_dir,
+            mem=8
+        ).join()
+        return variant_filter_status
+
+
+class IndelsFiltration(GATK4Stage):
+    """Filter the Indels using generic Hard filter."""
+
+    def _run(self):
+        filter_array = [
+            'QD < 2.0',
+            'FS > 200.0',
+            'ReadPosRankSum < -20.0',
+            'SOR > 10.0'
+        ]
+        filters = "'" + ' || '.join(filter_array) + "'"
+        var_filter_command = self.gatk_cmd('VariantFiltration', self.hard_filtered_indels_vcf)
+        var_filter_command += " -V " + self.raw_indels_vcf
+        var_filter_command += " --filter-expression " + filters
+        var_filter_command += " --filter-name 'INDEL_HARD_FILTER'"
+        variant_filter_status = executor.execute(
+            var_filter_command,
+            job_name='indel_filtration',
+            working_dir=self.exec_dir,
+            mem=16
+        ).join()
+        return variant_filter_status
+
+
+class MergeVariants(GATK4Stage):
+    """Merge multiple vcf files into one."""
+
+    vcf_files = ListParameter()
+    output_vcf_file = Parameter()
+
+    def _run(self):
+        # get the name of the list from the name of the first vcf to avoid overwriting another file
+        base_name = os.path.splitext(os.path.basename(self.vcf_files[0]))[0]
+        vcf_list = os.path.join(self.job_dir, base_name + '.list')
+        with open(vcf_list, 'w') as open_file:
+            for f in self.vcf_files:
+                open_file.write(f + '\n')
+        cmd = self.gatk_picard_cmd('MergeVcfs', self.output_vcf_file, input=vcf_list, reference=None)
+        return executor.execute(
+            cmd,
+            job_name='merge_vqsr',
```

Comment:
We should make the job name more generic, since it's not necessarily always VQSR data

Reply:
Changed to `merge_vcf`